aten/src/ATen/cpu/vec/vec512/vec512_int.h

```diff
@@ -1163,6 +1163,99 @@ inline Vectorized<int8_t> Vectorized<int8_t>::le(const Vectorized<int8_t>& other
   return (*this <= other) & Vectorized<int8_t>(1);
 }
 
+template <bool left_shift>
+Vectorized<int8_t> inline shift_512_8(const Vectorized<int8_t>& a, const Vectorized<int8_t>& b) {
+  // No vector instruction for shifting int8_t, so emulating it instead.
+
+  // Control masks for shuffle operation, treating 512 bits as an
+  // array of 8-bit elements, and considering pairs of neighboring
+  // elements.  Specifially, a mask named "ctl_M_N" (M,N in [0,1], and
+  // M!=N) is set so that shuffle will move element with index M from
+  // input pair into element with index N in output pair, and element
+  // with index M in output pair will be set to all 0s.
+  __m512i ctl_0_1 = _mm512_set_epi8(62, 0x80, 60, 0x80, 58, 0x80, 56, 0x80,
```

Comment:
a very small nit: to be consistent, should be keep all input arguments all in decimal, or all in hex?

Nevertheless, great work! Thanks, @alexsamardzic! :)

Reply:
These are "control" bytes for shuffle instructions, the meaning of individual values is: copy into output a byte from shuffle input from index given by control byte, or write 0 byte into output if control byte starts with bit 1 - I used 0x80 byte for this case.  I just found it easier and more readable, while testing/debugging, to differentiate between these two cases if valid indices given in decimal, and control byte value selecting 0 kept as hex number.

Reply:
Oh, I see. Thanks for clarifying! :)